[Shipping Labels] Parse selected hazmat from API response

irfano opened the pull request:
<!-- Remember about a good descriptive title. -->

<!-- Id number of the Linear issue this PR addresses, e.g., WOOMOB-373. -->
Closes WOOMOB-1273

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
This fixes an issue that occurred when opening a shipment with a purchased label containing a hazmat selection. Since we didn’t parse `selected_hazmat` from the API response, it wasn’t displayed. We now parse it correctly, and the selection is shown as expected.

### Steps to reproduce
<!-- Step-by-step testing instructions. For new user flows, consider instead stating the goal of the workflow and see if your PR reviewer can accomplish the workflow without specific steps! -->
1. Go to Orders.
2. Select an order that is eligible to purchase shipping labels.
3. Tap Create shipping label button.
4. Select a hazmat option.
5. Select a Package.
6. Purchase a label.
7. Verify that hazmat selection is shown as expected.
8. Navigate back to order details.
9. Tap Create shipping label button again.
10. Verify that hazmat selection is shown as expected. 
11. Kill the app and relaunch.
12. Follow 1-7 steps.

### The tests that have been performed
<!-- To give the reviewer an idea of what could be missed in terms of testing -->

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->
|Before|After|
|-|-|
|<img width="1280" height="2856" alt="Screenshot_20250904_171726" src="https://github.com/user-attachments/assets/a3890006-21d2-436d-86f6-d748b8f77751" />|<img width="1280" height="2856" alt="after" src="https://github.com/user-attachments/assets/99b4107e-dfdd-4950-ad71-34bbf4241425" />|

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

irfano committed:
Add `hazmatCategory` to `WooShippingLabelEntity`

This commit introduces a new field `hazmatCategory` to the `WooShippingLabelEntity` class and updates the database schema version to 61.

irfano committed:
Add Hazmat to `StoredDataDTO`

irfano committed:
Add hazmat category to ShippingLabelModel

irfano committed:
Set initial Hazmat selection

This commit initializes the `hazmatStatesFlow` with the Hazmat selection from the existing shipments when the ViewModel is created.
This ensures that if there are pre-existing labels with Hazmat categories, their selection is reflected in the UI.

irfano committed:
Map hazmatCategory in WooShippingLabelRepository

The `hazmatCategory` field is now mapped from the cached label data when a shipping label is voided.
This ensures that the `hazmatCategory` information is preserved after the void operation, as the API does not return this field.

irfano committed:
Add unit test for initial hazmat selection

dangermattic commented:
<!--
  0 Errors
  2 Warnings: This PR is larger than 300 lin...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/267">23.2</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

irfano committed:
Add release note for shipping labels fix

A release note is added for the fix related to displaying incorrect hazardous material options for purchased shipping labels.

hichamboushaba reviewed (approved):
Nice work @irfano, it works as expected.
